engine/src/main/java/org/camunda/bpm/engine/impl/cmd/SignalEventReceivedCmd.java

```diff
@@ -119,11 +120,11 @@ protected void sendSignalToExecution(CommandContext commandContext, String signa
 
     ExecutionManager executionManager = commandContext.getExecutionManager();
     ExecutionEntity execution = executionManager.findExecutionById(executionId);
-    ensureNotNull("Cannot find execution with id '" + executionId + "'", "execution", execution);
+    ensureNotNull(NotFoundException.class, "Cannot find execution with id '" + executionId + "'", "execution", execution);
```

Comment:
Previously, we threw a `NullValueException` here for the missing execution. If we change this, we should probably make that transparent (at least to the customer maybe). The latest discussion was around "no execution exists (anymore) =>NullValueException; no signal found (typo or already gone) => NotFoundException". Consolidating this probably makes sense, though. Let me know what you think.

Reply:
Is there any reference to that discussion and arguments for the current design? With other APIs it's inconsistent, however I do know that we throw `NotFoundException` usually when the entity adressed by a command does not exist. I have no strong motivation to change this behavior.

Reply:
This is the previous support case that introduced the `NullValueException` here (by introducing `ensureNotNull`): https://jira.camunda.com/browse/SUPPORT-13932?focusedCommentId=240528&page=com.atlassian.jira.plugin.system.issuetabpanels:comment-tabpanel#comment-240528